### PR TITLE
replace the white list from default mappings

### LINKED ISSUE #41756: Replace blacklist/whitelist in torch/quantization/default_mappings.py
The goal of this issue is to replace all instances of *whitelist* and *blacklist* in `torch/quantization/default_mappings.py`. See issue #41443 for more information.

### PR DESCRIPTION
**Pull request by VinodSKumar**
Replaced "whitelist" from default_mappings.py
Fixes #41756

### CODE REVIEW

**Comment by z-a-f** [comment]
Please, rebase :)

**Review by z-a-f — APPROVED** [review]
Will recheck after rebase

File: torch/quantization/quantize.py [hunk 1]
```diff
@@ -172,7 +172,7 @@ def add_quant_dequant(module):
         module._modules[name] = add_quant_dequant(child)
     return module
 
-def prepare(model, inplace=False, white_list=DEFAULT_QCONFIG_PROPAGATE_WHITE_LIST,
+def prepare(model, inplace=False, white_list=DEFAULT_QCONFIG_PROPAGATE_ALLOWED_LIST,
```
Comment (malfet) [inline torch/quantization/quantize.py:175]:
```suggestion
# Testing CI hooks
def prepare(model, inplace=False, white_list=DEFAULT_QCONFIG_PROPAGATE_ALLOWED_LIST,
```

File: torch/quantization/quantize.py [hunk 2]
```diff
@@ -172,6 +172,7 @@ def add_quant_dequant(module):
         module._modules[name] = add_quant_dequant(child)
     return module
 
+# Testing CI hooks
```
Comment (malfet) [inline torch/quantization/quantize.py:175]: Reverting temp change
```suggestion
```

**Comment by VinodSKumar** [comment]
@z-a-f rebase has been done

**Comment by VinodSKumar** [comment]
@z-a-f rebased again to since one CI was failing due to base env.

**Comment by VinodSKumar** [comment]
Resolved new conflicts and rebased

**Comment by VinodSKumar** [comment]
Rebased again to resolve conflicts

File: torch/quantization/_numeric_suite.py [hunk 3]
```diff
@@ -409,7 +409,7 @@ def prepare_model_outputs(
     float_module,
     q_module,
     Logger=OutputLogger,
-    white_list=DEFAULT_NUMERIC_SUITE_COMPARE_MODEL_OUTPUT_WHITE_LIST,
+    white_list=DEFAULT_NUMERIC_SUITE_COMPARE_MODEL_OUTPUT_ALLOWED_LIST,
```
Comment (malfet) [inline torch/quantization/_numeric_suite.py:412]: Forgot to rename this argument
```suggestion
    allow_list=DEFAULT_NUMERIC_SUITE_COMPARE_MODEL_OUTPUT_ALLOWED_LIST,
```

File: torch/quantization/_numeric_suite.py [hunk 4]
```diff
@@ -437,7 +437,7 @@ def compare_model_outputs(
     q_model,
     *data,
     Logger=OutputLogger,
-    white_list=DEFAULT_NUMERIC_SUITE_COMPARE_MODEL_OUTPUT_WHITE_LIST,
+    white_list=DEFAULT_NUMERIC_SUITE_COMPARE_MODEL_OUTPUT_ALLOWED_LIST,
```
Comment (malfet) [inline torch/quantization/_numeric_suite.py:440]: This needs to be renamed as well
```suggestion
    allow_list=DEFAULT_NUMERIC_SUITE_COMPARE_MODEL_OUTPUT_ALLOWED_LIST,
```

**Review by malfet — CHANGES_REQUESTED** [review]
You've forgot to rename the argument name
Also, you still have the merge conflicts (please feel free to post git commands you are using to rebase your change here)

**Comment by VinodSKumar** [comment]
> You've forgot to rename the argument name
> Also, you still have the merge conflicts (please feel free to post git commands you are using to rebase your change here)

@malfet This PR was for the issue [41756](https://github.com/pytorch/pytorch/issues/41756) relating to default_mappings.py, I presumed that it should only be related to it. Thanks for the feedback will update the argument and its usage.

 Regarding the rebase steps, please find the history of git commands used in this branch, Please let me know if any more information is required.
git checkout remove_black_white_list 
git fetch upstream
git rebase upstream/master
git add torch/quantization/quantize.py 
git add torch/testing/_internal/common_quantization.py 
git commit -m "Resolve conflicts from master"
git rebase --abort //due to the commit, the changes were not there in the staging area, hence restarted the rebasing.
git fetch upstream
git rebase upstream/master
git add torch/quantization/quantize.py 
git add torch/testing/_internal/common_quantization.py 
git rebase --continue 
git push origin remove_black_white_list -f

// second time rebasing for new conflicts, usually below are the steps for rebasing
git fetch upstream
git rebase upstream/master
git add torch/testing/_internal/common_quantization.py 
git rebase --continue
git push origin remove_black_white_list -f

**Comment by z-a-f** [comment]
LGTM, will leave to @malfet for landing

**Review by z-a-f — APPROVED** [review]
LGTM

**Review by malfet — APPROVED** [review]

**Comment by VinodSKumar** [comment]
Rebased to double check the PR changes

**Pull request closed without merging**